### PR TITLE
python3Packages.django-cachalot: init at 2.5.3

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Part of the effort of merging this PR https://github.com/NixOS/nixpkgs/pull/236877

Initial work of packaging this python module done by @RaitoBezarius . Enabled some extra tests

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Same here, changes look good, how about a nix-update-script to keep this up to date ?

File: pkgs/development/python-modules/django-cachalot/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/django-cachalot/default.nix:47]: Redundant.

https://github.com/noripyt/django-cachalot/blob/master/runtests.py#L8

```suggestion
```

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by mweinelt** [comment]
> Same here, changes look good, how about a nix-update-script to keep this up to date ?

Every python module has an update script by default.

**Comment by happysalada** [comment]
Didnt know this, thanks !

Then this looks good to merge, does anyone have any other concerns ?

**Comment by mweinelt** [comment]
I have an open change request.

**Comment by RaitoBezarius** [comment]
ofborg-eval didn't even finish, no approval, force push at the last minute.
image

I am increasingly concerned about this strange behavior in nixpkgs, I am dumbfounded.

**Comment by happysalada** [comment]
I understand your concern here, but i understand the will to try moving something forward.

I think what happened is that there was one comment to be taken care of, he took care of it and merged.

Waiting for formal approval can sometimes take a bit of time and when you can only spend a weekend on contributing, it could delay a PR by a week.
The hard part about asking for an approval is that sometimes people are busy and never respond.

Maybe we could have a process of asking for confirmation after someone commented on a PR and a minimum wait of 48h before taking any action.
I know we talked about a week in a chat once, im trying to find a middle ground to not have too much frustration on the side of the contributors as well.
I think we also want to make it an enjoyable process for people submitting a contribution otherwise we will always be in short numbers against an increasingly growing amount of work.

Im just trying to brainstorm here and not making any formal suggestions. My intention is not to hurt anyones feeling but to have a nice atmosphere between very valuable contributors.

**Comment by RaitoBezarius** [comment]
> I understand your concern here, but i understand the will to try moving something forward.

Moving forwards towards what? This PR has no user, no usecase except easing the upstream of Baserow and dealing with such things is clearly not going to make the upstream faster.

> I think what happened is that there was one comment to be taken care of, he took care of it and merged.

Yes and? I wrote:

> ofborg-eval didn't even finish, no approval, force push at the last minute.

No attestation of testing, nothing, the difference with pushing to master is very small here.

> Waiting for formal approval can sometimes take a bit of time and when you can only spend a weekend on contributing, it could delay a PR by a week. The hard part about asking for an approval is that sometimes people are busy and never respond.
> Maybe we could have a process of asking for confirmation after someone commented on a PR and a minimum wait of 48h before taking any action. I know we talked about a week in a chat once, im trying to find a middle ground to not have too much frustration on the side of the contributors as well. I think we also want to make it an enjoyable process for people submitting a contribution otherwise we will always be in short numbers against an increasingly growing amount of work.
> 
> Im just trying to brainstorm here and not making any formal suggestions. My intention is not to hurt anyones feeling but to have a nice atmosphere between very valuable contributors.

I will repeat it again. Formal approval is **necessary** in _almost_ all the cases. The frustration concern and "slowness" concern are all orthogonal, collaborating with so many people cannot happen at the speed of light, it has to have rules and let time do its things.

The concern about work growing while not having enough contributors is interesting because this PR does not address any current in nixpkgs, it just creates more work for everyone and for me to stumble on this type of stuff again and again.

For example, here's OP have been asked to pick up and finish the work https://github.com/NixOS/nixpkgs/pull/238407 which was asked long time ago and was ignored. It has been 3 weeks, what do we do about this?

There are rules to make the contribution process enjoyable for **everyone** and this must not happen at the expense of others. Enough valuable contributors have been burned out by this very selfish behavior of wanting to have fast PRs merged.

**Comment by happysalada** [comment]
| Moving forwards towards what? This PR has no user, no usecase except easing the upstream of Baserow and dealing with such things is clearly not going to make the upstream faster.
this is a python package. Maybe the OP needed this package for someone else he wanted to contribute. Any package made available to everyone is a win IMO. it reduces the chance that people come to nixpkgs and leave because stuff is just not ready.

| For example, here's OP have been asked to pick up and finish the work https://github.com/NixOS/nixpkgs/pull/238407 which was asked long time ago and was ignored. It has been 3 weeks, what do we do about this?
Something not finish is not ideal, I can't disagree on that. But isn't it better to have something working than nothing at all ? I understand your frustration with having sub-par code, but on the other side is the frustration of the end user with having things that are missing. I don't think there is a perfect situation here, we have to balance speed with code cleanliness.

| I will repeat it again. Formal approval is necessary in almost all the cases. The frustration concern and "slowness" concern are all orthogonal, collaborating with so many people cannot happen at the speed of light, it has to have rules and let time do its things.
I don't think this is practical in the current state of things. There aren't enough people to review everything. You could end up waiting for months before having someone review. This is just not a good contributor experience. People end up not contributing because of not being able to find someone for review.
What I suggest is
- Open a PR and wait for a reasonable amount of time. If no one comes to review then just merge it.
- If someone has come then try to get their approval. If there isn't a response within a particular timeframe then merge.
The cases of people coming and never having time to circle back are too many. I don't think the case of people self merging is a minority at this point. If you are working on something no one is interested then it's a necessary.
Unless we define a better rule than the strict ideal, I don't think we can have everyone on board.

People loosing motivation because things don't move forward is a very real concern, and It's my opinion that we can't ignore this.

I would love a future where you can't merge without an approval and you get a review in 24h, I very much believe we can create this.

@onny please be careful in the future, when someone comments on a PR you have to be mindful of their opinion. I hope this doesn't reduce your enthusiasm for nixpkgs, thank you for all the work you have done.

@RaitoBezarius I thank you for all the time energy and effort you have put into nixpkgs. I'm afraid the ideal future we all have in mind is still a little further away. I hope you can have patience with everything that isn't ideal yet. Thank you as well for everything you have done so far. 

We are all in this together, we all have the best interest of nixpkgs at heart, just different ways of expressing it. Let's go toward a great nixpkgs!

**Comment by RaitoBezarius** [comment]
> | Moving forwards towards what? This PR has no user, no usecase except easing the upstream of Baserow and dealing with such things is clearly not going to make the upstream faster. this is a python package. Maybe the OP needed this package for someone else he wanted to contribute. Any package made available to everyone is a win IMO. it reduces the chance that people come to nixpkgs and leave because stuff is just not ready.

This is… literally written in the description of the initial PR. Making this package available to everyone is not necessarily a win, this makes large scale upgrades harder with no real win.

> | For example, here's OP have been asked to pick up and finish the work #238407 which was asked long time ago and was ignored. It has been 3 weeks, what do we do about this? Something not finish is not ideal, I can't disagree on that. But isn't it better to have something working than nothing at all ? I understand your frustration with having sub-par code, but on the other side is the frustration of the end user with having things that are missing. I don't think there is a perfect situation here, we have to balance speed with code cleanliness.

Ah yes, legal compliance risk is "better than having nothing at all", is that right? I think you are sorely missing the point, it's not even about speed or code cleanliness, which are two dimensions I don't even agree on.

> 
> | I will repeat it again. Formal approval is necessary in almost all the cases. The frustration concern and "slowness" concern are all orthogonal, collaborating with so many people cannot happen at the speed of light, it has to have rules and let time do its things. I don't think this is practical in the current state of things. There aren't enough people to review everything. You could end up waiting for months before having someone review. This is just not a good contributor experience. People end up not contributing because of not being able to find someone for review. What I suggest is
> 
> * Open a PR and wait for a reasonable amount of time. If no one comes to review then just merge it.
> * If someone has come then try to get their approval. If there isn't a response within a particular timeframe then merge.
>   The cases of people coming and never having time to circle back are too many. I don't think the case of people self merging is a minority at this point. If you are working on something no one is interested then it's a necessary.
>   Unless we define a better rule than the strict ideal, I don't think we can have everyone on board.
> 
> People loosing motivation because things don't move forward is a very real concern, and It's my opinion that we can't ignore this.

OK, do you have a source to prove this? Is there people out there not contributing to nixpkgs due to that? How did you collect this information? Stop pushing claims like those without a shrewd of evidence.

We **do** have sources on valuable contributors who are **quitting** nixpkgs because a lot of people in here think that their right to have their change "fast" in nixpkgs is more important than anything else.

If anything else, we should not ignore this much more.

> I would love a future where you can't merge without an approval and you get a review in 24h, I very much believe we can create this.

Pardon the brutality, I think this is disillusion, you are underestimating the amount of efforts to make such a large project works and it shows. Time is not that compressible, and having SLO on review times is mind-boggling.

---

Anyway, I lost enough time on that matter, I am not interested into debating further.

**Comment by Ma27** [comment]
Coming here from #238407: while it's OK for people to make mistakes - even bigger ones - I'd like to emphasize that my main problem here is that you seem to actively ignore people bringing up these concerns @onny : this is _at least_ the second time you were doing questionable stuff and at least acknowledging that (and refraining from repeating that in the future) would help immensely.

**Pull request merged**
